src/utils/url-utils.ts

```diff
@@ -0,0 +1,9 @@
+export function replaceUrlSearchParam(url: URL, key: string, values: string[]) {
+	const searchParams = url.searchParams;
+	searchParams.delete(key);
+	for (const value of values) {
+		searchParams.append(key, value);
+	}
+	url.search = searchParams.toString();
+	return url;
+}
```

Comment:
not sure if I understand this. Why not use `searchParams.set(key, values)`?

Reply:
because of lists in the search params, e.g. 

`map?treeAge=young&treeAge=medium` 

`searchParams.set` would set the treeAge to one specific value, but we want multiple values with the same key. therefore we have to delete the key first and then add multiple values of the same key again.